### PR TITLE
feat: add TableHeaderCell type

### PR DESCRIPTION
**Pull request by djagya**
Adds a new type: `TableHeaderCell` similar to the `TableCell`. Adds a wrapper `TableHeaderRow` (without adding it to blocks) to enforce the "Table header cells should only always be in the first row" requirement

### CODE REVIEW

**Review by z0al — DISMISSED** [review]
LGTM

**Comment by z0al** [comment]
Travis CI 🤦🏻‍♂️

**Comment by DanweDE** [comment]
@djagya I didn't see anything satisfying this constraint:

> Table header cells should only always be in the first row

Ideally, we would check that if there is a header cell, it's in the first row, and if there's one header cell, all cells of the first row should be header cells.

File: .gitignore [hunk 1]
```diff
@@ -40,3 +40,4 @@
 
 **/.rpt2_cache
 .vscode/settings.json
+.idea
```
Comment (cezarsmpio) [inline .gitignore:43]: Shouldn't this be part of a global gitignore?

**Review by cezarsmpio — APPROVED** [review]
LGTM

File: .gitignore (hunk 1 above)
Comment (djagya) [reply .gitignore:43]: Idk, we add the same line for .vscode. Plus then other devs working on this project in IDEA won't have the same problem of files being included in git

**Comment by djagya** [comment]
@DanweDE now there's `TableHeaderRow` which is just a way to differentiate a row with cells and a row with header cells. Do I need to add it to the `BLOCKS` [enum](https://github.com/contentful/rich-text/blob/master/packages/rich-text-types/src/blocks.ts#L27)? I guess I should considering this make a new container type that stores only header cells?

**Comment by DanweDE** [comment]
> Do I need to add it to the BLOCKS enum?

The `table-header-cell` yes, it's an official new block node type :rocket: 
But don't add any header row type :scream: if that's your question. I think `TableHeaderRow` as you've implemented it seems like a good solution to enforce either a table without header cells or header cells only in the first row.

**Comment by djagya** [comment]
Yeah, that's what I was asking: are we fine with that `TableHeaderRow` working as an enforcement on the types level, but without adding it to blocks. You answered that, thank you!

File: .gitignore (hunk 1 above)
Comment (DanweDE) [reply .gitignore:43]: Personally, I don't have this problem as it's in my global gitignore like suggested by Cezar :shrug: 

I don't think it matters, although, in an ideal world .gitignore should be reserved for byproducts related to the repo.
If we don't add it then let's also remove `.vscode` :stuck_out_tongue:

**Review by z0al — APPROVED** [review]
Looks fine to me 👍🏻 

Also TIL about: `typescript-json-schema` package 😍 I will definitely use it in some of my projects

**Pull request merged**